### PR TITLE
Reduce G-API tests number

### PR DESCRIPTION
**Pull request by dbudnikov**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [ ] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

```
Xforce_builders=Custom,Custom Win,Custom Mac
build_gapi_standalone:Linux x64=ade-0.1.1f
build_gapi_standalone:Win64=ade-0.1.1f
build_gapi_standalone:Mac=ade-0.1.1f
build_gapi_standalone:Linux x64 Debug=ade-0.1.1f

Xbuild_image:Custom=centos:7
Xbuildworker:Custom=linux-1
build_gapi_standalone:Custom=ade-0.1.1f

build_image:Custom=ubuntu-openvino-2021.4.1:20.04
build_image:Custom Win=openvino-2021.4.1
build_image:Custom Mac=openvino-2021.4.1

test_modules:Custom=gapi,python2,python3,java
test_modules:Custom Win=gapi,python2,python3,java
test_modules:Custom Mac=gapi,python2,python3,java

buildworker:Custom=linux-1
# disabled due high memory usage: test_opencl:Custom=ON
test_opencl:Custom=OFF
test_bigdata:Custom=1
test_filter:Custom=*
```

### CODE REVIEW

File: modules/gapi/test/cpu/gapi_core_tests_fluid.cpp [hunk 1]
```diff
@@ -19,8 +19,6 @@ namespace opencv_test
 INSTANTIATE_TEST_CASE_P(MathOpTestFluid, MathOpTest,
                         Combine(Values(CV_8UC3, CV_8UC1, CV_16SC1, CV_32FC1),
                                 Values(cv::Size(1920, 1080),
-                                       cv::Size(1280, 720),
-                                       cv::Size(640, 480),
```
Comment (dmatveev) [inline modules/gapi/test/cpu/gapi_core_tests_fluid.cpp:23]: This selection looks strange. Shouldn't the `1080` option be removed & `720` be kept as in the previous file?
As far I can see, the below cases also instantiate `720` and not `1080`.

File: modules/gapi/test/cpu/gapi_core_tests_fluid.cpp [hunk 2]
```diff
@@ -236,65 +206,55 @@ INSTANTIATE_TEST_CASE_P(ConvertToFluid, ConvertToTest,
...
                                        cv::Size(128, 128)),
                                 Values(CV_8UC3),
                                 Values(CORE_FLUID)));
 
 INSTANTIATE_TEST_CASE_P(Merge4TestFluid, Merge4Test,
                         Combine(Values(CV_8UC1),
                                 Values(cv::Size(1920, 1080),
-                                       cv::Size(1280, 720),
-                                       cv::Size(640, 480),
                                        cv::Size(128, 128)),
```
Comment (dmatveev) [inline modules/gapi/test/cpu/gapi_core_tests_fluid.cpp:244]: Maybe we should declare the list of resolutions we test one and for all the tests? Then `ValuesIn(...)` could be used.

**Review by dmatveev — APPROVED** [review]
If we still have cases with multiple resolutions, I'd propose to declare a constant vector of the sizes once and for all tests - and use `ValuesIn(...)` in the test instantiations

File: modules/gapi/test/cpu/gapi_core_tests_fluid.cpp (hunk 2 above)
Comment (dbudnikov) [reply modules/gapi/test/cpu/gapi_core_tests_fluid.cpp]: Done

File: modules/gapi/test/cpu/gapi_core_tests_fluid.cpp (hunk 1 above)
Comment (dbudnikov) [reply modules/gapi/test/cpu/gapi_core_tests_fluid.cpp]: Done

**Pull request merged**